### DISCUSSION TITLE
Code about getting mcmc sampling results

### DISCUSSION THREAD

**Discussion by 0batsy0** [phoebe-project/phoebe2#833] [Questions]
I hope to extract the data sampled by mcmc, so I used the following two different codes to try to obtain it, but the obtained results seem to be somewhat different. I don’t quite understand the difference between the functions of these two lines of code.Here are the distribution diagrams obtained by using these two writing methods for parameter q.
![1](https://github.com/phoebe-project/phoebe2/assets/130527283/c28e14e9-d9ea-4b60-b562-4e9219e33ca1)
![3](https://github.com/phoebe-project/phoebe2/assets/130527283/c5281641-41bf-4873-9fa4-08006531c20c)

![2](https://github.com/phoebe-project/phoebe2/assets/130527283/054d338c-5ad6-433f-b00b-bd7faa64ac33)
![4](https://github.com/phoebe-project/phoebe2/assets/130527283/2e93e617-5e4d-4352-bdf1-b783e30a5ce7)

**Comment by kecnry** [comment]
This is a good question and probably something we should add to the tutorials somewhere!

By passing `None` to all the keyword arguments, phoebe will use the default _proposed_ values in the solution for burning, thin, etc.  Can you see if `process_mcmc_chains_from_solution(tes, 'emcee_sol', burnin=0, thin=1, lnprob_cutoff=-np.inf)` gives you what you'd expect?

To see the default values being applied when passing `None`: `print(tes.filter(qualifier=['burnin', 'thin', 'lnprob_cutoff'], solution='emcee_sol'))`

**Reply by 0batsy0** [reply]
Thanks for your reply, I have successfully solved it!!!But I still don’t quite understand why when burnin exists by default (for example, I checked that the value is 226), the two methods will get the same number of data (my understanding is that the relative number of burnin should be subtracted)

**Reply by kecnry** [reply]
I would expect the shape/length to change as well.  If you call `.shape` on each of the returned arrays, before plotting, what do they say?

**Reply by 0batsy0** [reply]
Thank you!I found the problem!
